src/cloudsc_fortran_pyiface/pythonsrc/cloudsc_data.py

```diff
@@ -0,0 +1,382 @@
+
+    for argname in argnames_nlev:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev  ,nblocks), order='F')
+
+    for argname in argnames_nlevp:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev+1,nblocks), order='F')
+
+    for argname in argnames_buffer:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev,3+NCLV,nblocks), order='F')
+
+    for argname in argnames_tend:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev,nblocks), order='F')
+
+    for argname in argnames_tend_cld:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nlev,NCLV,nblocks), order='F')
+
+
+    for argname in argnames_nproma:
+        print('Define null field:',argname)
+        fields[argname] = np.zeros(shape=(nproma,nblocks), order='F')
+
+    fields['ydomcst']=clsc.yomcst.TOMCST()
+    fields['ydoethf']=clsc.yoethf.TOETHF()
+    fields['ydecldp']=clsc.yoecldp.TECLDP()
+    fields['ydephli']=clsc.yoephli.TEPHLI()
+
+    return fields
+
+def load_input_fortran_fields(path, nproma, nlev, nblocks, fields):
+    """
+    load_input_fortran_fields returns:
+    - set of variables needed to initiate computation of the Fortran kernel.
+    """
+    argnames_nlev = [
+        'pt', 'pq',
+        'pvfa', 'pvfl', 'pvfi', 'pdyna', 'pdynl', 'pdyni',
+        'phrsw', 'phrlw','pvervel','pap','plu','plude',
+        'psnde', 'pmfu', 'pmfd',
+        'pa', 'psupsat',
+        'plcrit_aer','picrit_aer','pre_ice',
+        'pccn', 'pnice'
+    ]
+    argnames_nlevp = [
+        'paph'
+    ]
+
+    argnames_withnclv= [
+        'pclv','tendency_tmp_cld'
+    ]
+
+    argnames_tend = [
+        'tendency_tmp_t','tendency_tmp_q','tendency_tmp_a'
+    ]
+
+    argnames_scalar = [
+         'kfldx'
+    ]
+
+    argnames_nproma = [
+        'plsm', 'ldcum', 'ktype'
+    ]
+
+
+    with h5py.File(path, 'r') as f:
+        fields['KLON'] = f['KLON'][0]
+        fields['KLEV'] = f['KLEV'][0]
+        fields['PTSPHY'] = f['PTSPHY'][0]
+
+        for argname in argnames_nlev:
+            print('Loading field:',argname)
+            fields[argname] = np.asfortranarray(
+                                 np.transpose(
+                                    np.reshape(
+                                       np.ascontiguousarray(f[argname.upper()]),
+                                       (nblocks,nlev,nproma),order='C')))
+
+        for argname in argnames_nlevp:
+            print('Loading field:',argname)
+            fields[argname] = np.asfortranarray(
+                                 np.transpose(
+                                    np.reshape(
+                                       np.ascontiguousarray(f[argname.upper()]),
+                                       (nblocks,nlev+1,nproma),order='C')))
+
+        for argname in argnames_withnclv:
```

Comment:
I personally think the dynamic inflection here is almost over-engineered. Simply moving the casting into a utility routine, or doing it as part of an "allocate" or "load" helper routine would allow you to simply list the fields here. However, this comes down primarily to taste, so it doesn't need to get addressed here - just a comment.